tokio/src/io/split.rs

```diff
@@ -54,13 +52,19 @@ cfg_io_util! {
 }
 
 struct Inner<T> {
-    locked: AtomicBool,
-    stream: UnsafeCell<T>,
+    stream: Mutex<UnsafeCell<T>>,
```

Comment:
When you add a mutex, you don't also need an `UnsafeCell`.

Reply:
Ah you're right.